tools/googlesearch/searchTool.go

```diff
@@ -0,0 +1,48 @@
+package googlesearch
+
+import (
+	"strings"
+
+	"github.com/tmc/langchaingo/tools"
+	"github.com/tmc/langchaingo/utils/serpapi"
+)
+
+type SearchTool struct {
+	name        string
+	description string
+	serpapi     *serpapi.SerpapiWrapper
+}
+
+func NewSearchTool() (tools.Tool, error) {
+	serpapi, err := serpapi.NewSerpapiWrapper()
+	if err != nil {
+		return nil, err
+	}
+	return &SearchTool{
+		name: "Google Search",
+		description: `"A wrapper around Google Search. "
+        "Useful for when you need to answer questions about current events. "
+		"Always one of the first options when you need to find information on internet"
+        "Input should be a search query."
+		"Priority: 1"`,
+		serpapi: serpapi,
+	}, nil
+
+}
+
+func (s *SearchTool) Name() string {
```

Comment:
there's no reason in Go to wrap a field like this, in general. Instead you should just export the field and avoid writing a method in this manner.